src/scheduler/tasks/SynchronizeTagsImportTask.ts

```diff
@@ -0,0 +1,75 @@
+import Constants from '../../utils/Constants';
+import { ImportStatus } from '../../types/GlobalType';
+import { LockEntity } from '../../types/Locking';
+import LockingManager from '../../locking/LockingManager';
+import Logging from '../../utils/Logging';
+import SchedulerTask from '../SchedulerTask';
+import { ServerAction } from '../../types/Server';
+import Tag from '../../types/Tag';
+import TagStorage from '../../storage/mongodb/TagStorage';
+import TagValidator from '../../server/rest/v1/validator/TagValidation';
+import { TaskConfig } from '../../types/TaskConfig';
+import Tenant from '../../types/Tenant';
+
+const MODULE_NAME = 'SynchronizeUsersImportTask';
+
+export default class SynchronizeTagsImportTask extends SchedulerTask {
+  async processTenant(tenant: Tenant, config: TaskConfig): Promise<void> {
+    const synchronizeTagsImport = LockingManager.createExclusiveLock(tenant.id, LockEntity.TAGS, 'synchronize-tags-import');
+    if (await LockingManager.acquire(synchronizeTagsImport)) {
+      try {
+        const importedTags = await TagStorage.getImportedTags(tenant.id, { statuses: [ImportStatus.UNKNOWN] }, Constants.DB_PARAMS_MAX_LIMIT);
```

Comment:
If you have 1 million tags to import what do you think will happen to the VM's memory in prod?
Use pagination constants like EXPORT_RECORD_MAX_COUNT and process tags per block.
Moreover the count will be -1 in case of result > 500.

Reply:
Done